test/unit/state/hal-forms.ts

```diff
@@ -54,6 +54,41 @@ describe('HAL forms', () => {
 
     expect(deleteAction).to.eql(expectedDeleteAction);
 
+  });
+  it('should relate relative URIs of embedded HAL forms to the embedded resource\'s self uri', async () => {
```

Comment:
Tests added to #393 verify the behaviour when no `target` is provided. 
Could a test covering the missing `target` use case be added to this PR? :)

Reply:
Done